Update error checking/handle finicky API

taers232c opened the pull request.

jay0lee commented on src/gam.py:
```diff
@@ -7413,6 +7416,9 @@ def doUpdateVaultMatter(action=None):
   if body:
     print u'Updating matter %s...' % sys.argv[3]
+    # bah, API requires name to be sent on update even when it's not changing
+    result = callGAPI(v.matters(), u'get', matterId=matterId, view=u'BASIC')
```
if body['name'] is already set we shouldn't do an expensive API call, make this conditional on name not being set.

taers232c commented:
Done
--
ross.scroggs@gmail.com



On Jul 29, 2017, at 3:44 AM, Jay Lee <notifications@github.com> wrote:

> @jay0lee commented on this pull request.
> 
> In src/gam.py:
> 
> > @@ -7413,6 +7416,9 @@ def doUpdateVaultMatter(action=None):
>      action_kwargs = {}
>    if body:
>      print u'Updating matter %s...' % sys.argv[3]
> +    # bah, API requires name to be sent on update even when it's not changing
> +    result = callGAPI(v.matters(), u'get', matterId=matterId, view=u'BASIC')
> if body['name'] is already set we shouldn't do an expensive API call, make this conditional on name not being set.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

taers232c commented:
Jay,

Are there any issues here that I need to address?

Ross

taers232c commented:
def doActionVaultMatter(action):

You already have: gam delete hold and gam undelete hold, gam update hold
Add these (gam close hold and gam reopen hold) so its always gam action hold

taers232c commented:
Jay,

Are there issues I still need to deal with here?

Thanks,

Ross

jay0lee commented:
I don't see the need to split action vs update for Vault.

taers232c commented:
A: Split update/action - consistent, user has to learn one form
    all commands are verb object
    (As a bonus, it matches the API)

gam update vaultmatter|matter <MatterItem> [name <String>] [description
<string>]

        [addcollaborator|addcollaborators <EmailItemList>]
[removecollaborator|removecollaborators <EmailItemList>]

gam close vaultmatter|matter <MatterItem>

gam reopen vaultmatter|matter <MatterItem>

gam delete vaultmatter|matter <MatterItem>

gam undelete vaultmatter|matter <MatterItem>


B: Semi-split update/action - inconsistent, user has to learn two forms
    update is verb object
    delete|undelete are verb object or update object action delete|undelete
    reopen|close are update object action reopen|close

gam update vaultmatter|matter <MatterItem> [name <String>] [description
<string>] [action close|reopen|delete|undelete]

        [addcollaborator|addcollaborators <EmailItemList>]
[removecollaborator|removecollaborators <EmailItemList>]

gam delete vaultmatter|matter <MatterItem>

gam undelete vaultmatter|matter <MatterItem>


In your version you do:

1: Execute update

2: Execute close|reopen|delete|undelete

3: Update collaborators


If the current state is OPEN and the action is close: 1 works, 2 works, 3
fails because matter is now CLOSED

If the current state is CLOSED and the action is open: 1 fails because the
state is CLOSED, 2 works, 3 works

If the current state is CLOSED and the action is delete: 1 fails because
the state is CLOSED, 2 works, 3 fails because matter is now DELETED

If the current state is DELETED and the action is reopen: 1 fails because
the state is DELETED, 2 works, 3 fails because matter is now CLOSED


With B you're giving the user the impression that you can make changes and
perform a state change action at the same time; you can't.

With A, you can do an update or perform a state change action, no confusion.


I'm disregarding state change actions that are not appropriate to the
current state as they will fail in A and B; it's a different issue not
related to form.


On Thu, Aug 10, 2017 at 1:37 PM, Jay Lee <notifications@github.com> wrote:

> I don't see the need to split action vs update for Vault.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jay0lee/GAM/pull/543#issuecomment-321666295>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIU8L4diank3UcLhzU3gon25E52Ss5Nqks5sW2n1gaJpZM4OnLFr>
> .
>



-- 
Ross Scroggs
ross.scroggs@gmail.com
